src/frontend/src/lib.rs

```diff
@@ -65,41 +66,31 @@ use clap::Parser;
 use pgwire::pg_server::pg_serve;
 use session::SessionManagerImpl;
 
+/// CLI arguments received by meta node. Overwrites fields in
+/// [`risingwave_common::config::FrontendConfig`].
 #[derive(Parser, Clone, Debug)]
 pub struct FrontendOpts {
-    // TODO: rename to listen_address and separate out the port.
-    #[clap(long, default_value = "127.0.0.1:4566")]
-    pub host: String,
+    #[clap(long = "host")]
+    pub listen_addr: Option<String>,
 
-    // Optional, we will use listen_address if not specified.
     #[clap(long)]
     pub client_address: Option<String>,
 
-    // TODO: This is currently unused.
     #[clap(long)]
-    pub port: Option<u16>,
+    pub meta_addr: Option<String>,
```

Comment:
Should `meta_addr` be a shared config instead of being under the `[frontend]` section?

Reply:
It's for backward compatibility 🤣

Reply:
The limitation of sharing clap and serde in the same struct is that the CLI args can only be the subset of one `XxxConfig`. If we, say, put `meta_addr` in `[server]`, then CLI should both overwrite `FrontendConfig` and `ServerConfig`. I guess in that case we would still need to implement a proc macro on our own?